DXRModules/DeusEx/Classes/DXRFixup.uc

```diff
@@ -797,6 +806,37 @@ static function FixConversationAddNote(Conversation c, string textSnippet)
     }
 }
 
-defaultproperties
+function SetLampState(#var(prefix)Lamp lmp, bool turnOn) // could maybe go in DXRLamp if it existed
+{
+    if (turnOn) {
+        lmp.bOn = True;
+        lmp.LightType = LT_Steady;
+        // lmp.PlaySound(sound'Switch4ClickOn');
+        lmp.bUnlit = True;
+        lmp.ScaleGlow = 3.0;
+    } else if (lmp.bOn) {
+        lmp.bOn = False;
+        lmp.LightType = LT_None;
+        // lmp.PlaySound(sound'Switch4ClickOff');
+        lmp.bUnlit = False;
+        lmp.ResetScaleGlow();
+    }
+}
+
+function SetAllLampsState(bool type1, bool type2, bool type3, optional Vector loc, optional float rad, optional bool applyToModifiedMaps)
 {
+    local #var(prefix)Lamp lmp;
+
+    if (class'MenuChoice_AutoLamps'.default.enabled == false || (!applyToModifiedMaps && #defined(revision || gmdx)))
+        return;
+
+    if (rad == 0.0) {
+        foreach AllActors(class'#var(prefix)Lamp', lmp) {
+            SetLampState(lmp, (Lamp1(lmp) != None && type1) || (Lamp2(lmp) != None && type2) || (Lamp3(lmp) != None && type3));
+        }
+    } else {
+        foreach RadiusActors(class'#var(prefix)Lamp', lmp, rad, vectm(loc.x, loc.y, loc.z)) {
```

Comment:
surprised you're calling vectm here instead of in the callers, but if you want to do it here maybe this is better
```suggestion
        foreach RadiusActors(class'#var(prefix)Lamp', lmp, rad, loc*coords_mult) {
```